thumbnail/index.php

```diff
@@ -5,27 +5,119 @@
 use Directus\Util\ArrayUtils;
 use Directus\Filesystem\Thumbnailer as ThumbnailerService;
 
-try {
+try 
+{
     $app = \Directus\Bootstrap::get('app');
-
-    // if the thumb already exists, return it
+    $config = $app->container->get('config')->get('thumbnailer');
+    
+    /*
+	    Set the fallback option in the thumbnailer config - '/api/configuration.php'
+	    If not set, a 404 img will be sent back if the dimensions 
+        are not supported within 'supportedThumbnailDimensions'
+    */
+    
+    if (ArrayUtils::get($config, 'fallback'))
+    {
+		/*
+			Fall back to one of supported sizes
+		*/
+	    
+	    $pathinfo = $app->request->getPathInfo();
+	    $pathinfo = ltrim($pathinfo, '/');
+	    $pathinfo = explode('/', $pathinfo);
+	    
+	    $config = $app->container->get('config')->get('thumbnailer');
+	    $sizes = ArrayUtils::get($config, 'supportedThumbnailDimensions');
+	    $redirect = NULL;
+	    
+	    /*
+		    Get the nearest size above the required size
+		    We must assume the first and second indexes are the width and height
+	    */
+	    
+	    foreach ($sizes as $size)
+	    {
+		    $size = explode('x', trim($size));
+		    
+		    $width = (int) $size[0];
+		    $height = (int) $size[1];
+		    
+		    $imgwidth = (int) $pathinfo[0];
+		    $imgheight = (int) $pathinfo[1];
+		    
+		    /*
+			    If the sizes are validated continue with the rest of the script
+			    Otherwise falback to the nearest valid sizes
+		    */
+		    
+		    if ($imgwidth === $width && $imgheight === $height)
+		    {
+			    break;
+		    }
+		    elseif ($imgwidth < $width && $imgheight < $height)
```

Comment:
Does the intention of this is to work with sizes above the requested size? for example if the requested size is `200x200`, but it's configured to only allowed `100x100` and fallback enabled, this won't redirect to `100x100`, because `200` is not less than `100`. but `50x50` will redirect to `100x100`.

Reply:
I did make some updates that will work with sizes under and above, but not sure how to update this current pull request, should I make another?

Reply:
@philleepflorence If you push to the same branch, it gets added to the PR :)

Reply:
Okay, will do later this evening, thanks!